plugins/pencilblue/controllers/admin/sites/site_form.js

```diff
@@ -75,22 +75,36 @@ module.exports = function SiteFormModule(pb) {
     var options = {isNew: true};
     //todo:: refactor angular on site_form to remove unneeded properties
     setupAngularObj(self, options, cb);
-
   };
 
   function setupAngularObj(self, options, cb){
     var isNew = options.isNew,
       display = options.display,
       host = options.host,
+      supportedLocales = {},
+      savedLocales = options.savedLocales || {},
+      selectedLocales = [],
+      defaultLocale = options.defaultLocale || "",
       isActive = options.isActive,
       uid = options.uid;
 
+    for (var locale in pb.Localization.supportedLookup) {
```

Comment:
Apologies if this is a bit confusing. I tried to make variables that make sense in the context they're being used. "savedLocales" is populated from the "supportedLocales" object pulled from the DB. "selectedLocales" is used to populate the drop-down and thus needs to be an array. Basically, one's an object, one's an array.

I use this iteration over Localization's supportedLookup to accomplish a few goals:
1. Filter out locales previously saved that may no longer be valid
2. Populate the "supportedLocales" object with true/false values pulled from "savedLocales"(DB)
3. Populate "selectedLocales" with all valid "savedLocales"(DB)

I could use the approach above, but I'd still have to iterate over Localization's supportedLocales to populate the supportedLocales object.